### PR TITLE
[12.0][ADD] - substitution rules for transmit method

### PR DESCRIPTION
**Pull request by sbejaoui**
Sometimes a transmission method defined by default for the partner is not
applicable. For example, when the transmission method is e-mail and the
partner's e-mail is not set. This addon allows you to define substitution
rules when an invoice matches certain criteria.

This module can also be used when it is not easy to define the partner's
preference for the transmission method or the partner's preference is not
legally permitted. For example, in some countries, invoices with payment
direct debit method must be sent to the customer by post.

### CODE REVIEW

**Comment by sbejaoui** [comment]
cc/ @gva-acsone

**Review by gva-acsone — APPROVED** [review]
Functional test

File: account_invoice_transmit_method_substitution_rule/demo/transmit_method_substitution_rule.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+<odoo noupdate="1">
+
+    <record model="transmit.method.substitution.rule" id="transmit_method_substitution_rule_demo_1">
+        <field name="name">Transmit method Post for partners with undefined email</field>
+        <field name="domain">[('partner_id.email', '=', False)]</field>
+        <field name="transmit_method_id" eval="ref('account_invoice_transmit_method.post')"/>
+    </record>
+
+    <record model="transmit.method.substitution.rule" id="transmit_method_substitution_rule_demo_2">
+        <field name="name">Transmit method Post for partners with undefined email</field>
```
Comment (qgroulard) [inline account_invoice_transmit_method_substitution_rule/demo/transmit_method_substitution_rule.xml]:
```suggestion
        <field name="name">Transmit method Email for partners with incomplete address</field>
```

**Review by qgroulard — APPROVED** [review]
LGTM (code review)

**Comment by sbejaoui** [comment]
@OCA/core-maintainers , This PR is ready to merge

**Pull request merged**